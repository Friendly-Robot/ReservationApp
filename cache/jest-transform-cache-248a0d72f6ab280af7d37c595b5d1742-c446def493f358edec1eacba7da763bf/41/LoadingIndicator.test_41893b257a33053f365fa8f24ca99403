514458c26cabb3c6b5342fc5fa473301
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

require("react-native");

var _react = _interopRequireDefault(require("react"));

var _LoadingIndicator = _interopRequireDefault(require("../LoadingIndicator"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

describe('<LoadingIndicator />', function () {
  it('should be defined', function () {
    expect(_LoadingIndicator.default).toBeDefined();
  });
  it('renders correctly', function () {
    var tree = _reactTestRenderer.default.create(_react.default.createElement(_LoadingIndicator.default, null)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkxvYWRpbmdJbmRpY2F0b3IudGVzdC5qcyJdLCJuYW1lcyI6WyJkZXNjcmliZSIsIml0IiwiZXhwZWN0IiwiTG9hZGluZ0luZGljYXRvciIsInRvQmVEZWZpbmVkIiwidHJlZSIsInJlbmRlcmVyIiwiY3JlYXRlIiwidG9KU09OIiwidG9NYXRjaFNuYXBzaG90Il0sIm1hcHBpbmdzIjoiOztBQUlBOztBQUNBOztBQUNBOztBQUdBOztBQUVBQSxRQUFRLENBQUMsc0JBQUQsRUFBeUIsWUFBTTtBQUNyQ0MsRUFBQUEsRUFBRSxDQUFDLG1CQUFELEVBQXNCLFlBQU07QUFDNUJDLElBQUFBLE1BQU0sQ0FBQ0MseUJBQUQsQ0FBTixDQUF5QkMsV0FBekI7QUFDRCxHQUZDLENBQUY7QUFJQUgsRUFBQUEsRUFBRSxDQUFDLG1CQUFELEVBQXNCLFlBQU07QUFDNUIsUUFBTUksSUFBSSxHQUFHQywyQkFBU0MsTUFBVCxDQUFnQiw2QkFBQyx5QkFBRCxPQUFoQixFQUFzQ0MsTUFBdEMsRUFBYjs7QUFDQU4sSUFBQUEsTUFBTSxDQUFDRyxJQUFELENBQU4sQ0FBYUksZUFBYjtBQUNELEdBSEMsQ0FBRjtBQUlELENBVE8sQ0FBUiIsInNvdXJjZXNDb250ZW50IjpbIi8qKlxuICogQGZvcm1hdFxuICovXG5cbmltcG9ydCAncmVhY3QtbmF0aXZlJztcbmltcG9ydCBSZWFjdCBmcm9tICdyZWFjdCc7XG5pbXBvcnQgTG9hZGluZ0luZGljYXRvciBmcm9tICcuLi9Mb2FkaW5nSW5kaWNhdG9yJztcblxuLy8gTm90ZTogdGVzdCByZW5kZXJlciBtdXN0IGJlIHJlcXVpcmVkIGFmdGVyIHJlYWN0LW5hdGl2ZS5cbmltcG9ydCByZW5kZXJlciBmcm9tICdyZWFjdC10ZXN0LXJlbmRlcmVyJztcblxuZGVzY3JpYmUoJzxMb2FkaW5nSW5kaWNhdG9yIC8+JywgKCkgPT4ge1xuICBpdCgnc2hvdWxkIGJlIGRlZmluZWQnLCAoKSA9PiB7XG4gICAgZXhwZWN0KExvYWRpbmdJbmRpY2F0b3IpLnRvQmVEZWZpbmVkKCk7XG4gIH0pO1xuXG4gIGl0KCdyZW5kZXJzIGNvcnJlY3RseScsICgpID0+IHtcbiAgICBjb25zdCB0cmVlID0gcmVuZGVyZXIuY3JlYXRlKDxMb2FkaW5nSW5kaWNhdG9yIC8+KS50b0pTT04oKTtcbiAgICBleHBlY3QodHJlZSkudG9NYXRjaFNuYXBzaG90KCk7XG4gIH0pO1xufSlcblxuIl19