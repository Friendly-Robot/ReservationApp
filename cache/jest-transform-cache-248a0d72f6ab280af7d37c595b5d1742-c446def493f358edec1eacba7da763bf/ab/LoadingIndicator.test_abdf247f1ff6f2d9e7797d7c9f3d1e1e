20708a2e1f88b5ccb199bd659f529ff1
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

require("react-native");

var _react = _interopRequireDefault(require("react"));

var _LoadingIndicator = _interopRequireDefault(require("../LoadingIndicator"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

it('renders correctly', function () {
  var tree = _reactTestRenderer.default.create(_react.default.createElement(_LoadingIndicator.default, null)).toJSON();

  expect(tree).toMatchSnapshot();
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkxvYWRpbmdJbmRpY2F0b3IudGVzdC5qcyJdLCJuYW1lcyI6WyJpdCIsInRyZWUiLCJyZW5kZXJlciIsImNyZWF0ZSIsInRvSlNPTiIsImV4cGVjdCIsInRvTWF0Y2hTbmFwc2hvdCJdLCJtYXBwaW5ncyI6Ijs7QUFJQTs7QUFDQTs7QUFDQTs7QUFHQTs7QUFFQUEsRUFBRSxDQUFDLG1CQUFELEVBQXNCLFlBQU07QUFDNUIsTUFBTUMsSUFBSSxHQUFHQywyQkFBU0MsTUFBVCxDQUFnQiw2QkFBQyx5QkFBRCxPQUFoQixFQUFzQ0MsTUFBdEMsRUFBYjs7QUFDQUMsRUFBQUEsTUFBTSxDQUFDSixJQUFELENBQU4sQ0FBYUssZUFBYjtBQUNELENBSEMsQ0FBRiIsInNvdXJjZXNDb250ZW50IjpbIi8qKlxuICogQGZvcm1hdFxuICovXG5cbmltcG9ydCAncmVhY3QtbmF0aXZlJztcbmltcG9ydCBSZWFjdCBmcm9tICdyZWFjdCc7XG5pbXBvcnQgTG9hZGluZ0luZGljYXRvciBmcm9tICcuLi9Mb2FkaW5nSW5kaWNhdG9yJztcblxuLy8gTm90ZTogdGVzdCByZW5kZXJlciBtdXN0IGJlIHJlcXVpcmVkIGFmdGVyIHJlYWN0LW5hdGl2ZS5cbmltcG9ydCByZW5kZXJlciBmcm9tICdyZWFjdC10ZXN0LXJlbmRlcmVyJztcblxuaXQoJ3JlbmRlcnMgY29ycmVjdGx5JywgKCkgPT4ge1xuICBjb25zdCB0cmVlID0gcmVuZGVyZXIuY3JlYXRlKDxMb2FkaW5nSW5kaWNhdG9yIC8+KS50b0pTT04oKTtcbiAgZXhwZWN0KHRyZWUpLnRvTWF0Y2hTbmFwc2hvdCgpO1xufSk7XG4iXX0=