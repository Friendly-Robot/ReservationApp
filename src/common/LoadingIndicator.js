import React from 'react';
import { ActivityIndicator, StyleSheet, Text, View } from 'react-native';

import theme from '../theme';

type props = {
  activityStyles: object,
  containerStyles: object,
  textStyles: object,
  textLabel: string,
};

export default LoadingIndicator = ({
  activityStyles,
  containerStyles,
  textStyles,
  textLabel,
}) => (
  <View style={[styles.container, containerStyles]}>
    <ActivityIndicator
      animating
      color={theme.COLOR_PRIMARY}
      size={'large'}
      style={activityStyles}
    />
    <Text style={[styles.text, textStyles]}>
      { textLabel }
    </Text>
  </View>
);

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
    paddingHorizontal: 25,
  },
  text: {
    color: theme.COLOR_PRIMARY,
    fontSize: theme.FONT_SIZE_MEDIUM,
    marginTop: 25,
  },
});