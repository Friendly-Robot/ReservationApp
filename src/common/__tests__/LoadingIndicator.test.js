/**
 * @format
 */

import 'react-native';
import React from 'react';
import LoadingIndicator from '../LoadingIndicator';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('<LoadingIndicator />', () => {
  it('should be defined', () => {
    expect(LoadingIndicator).toBeDefined();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<LoadingIndicator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})

