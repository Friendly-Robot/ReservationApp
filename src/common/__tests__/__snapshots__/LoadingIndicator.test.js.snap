// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 25,
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="rgb(30, 147, 242)"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    style={
      Array [
        Object {
          "color": "rgb(30, 147, 242)",
          "fontSize": 14,
          "marginTop": 25,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 25,
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="rgb(30, 147, 242)"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    style={
      Array [
        Object {
          "color": "rgb(30, 147, 242)",
          "fontSize": 14,
          "marginTop": 25,
        },
        undefined,
      ]
    }
  />
</View>
`;
